ggml/src/gguf.cpp

```diff
@@ -1284,14 +1304,14 @@ struct gguf_writer {
             GGML_ASSERT(info.t.data);
             memcpy(buf.data() + offset, info.t.data, nbytes);
         }
+        written_bytes += nbytes;
 
         pad(alignment);
     }
 };
 
-void gguf_write_to_buf(const struct gguf_context * ctx, std::vector<int8_t> & buf, bool only_meta) {
-    const struct gguf_writer gw(buf);
-
+template <typename Writer>
```

Comment:
```suggestion
template <typename writer_t>
```

Preferably use snake_case.